decentralised_rendering.md

```diff
@@ -10,7 +10,38 @@ Accepted
 
 The purpose of the decentralised document renderer is to allow OpenAttestation (OA) document issuers to style their documents without code change to the different implementations of the document viewer. It does so by embedding the website specified by the document issuers as an iframe or webview (for mobile apps) and sending the content of the OA document into the iframe.
 
-## Frame-to-Frame Communication
+It's important to note that a decentralised renderer can be configured to display multiple kind of certificates, known as `Layout`.
+
+The document viewer and the document renderer will rely on [postMessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) for the communication, and more specifically OpenAttestation rely on [penpal](https://github.com/Aaronius/penpal), a promise-based library for securely communicating with iframes via postMessage.
+
+## Definitions
+The following terms will be used across the document:
+
+- `Document Viewer`: Website that loads a `Decentralised Renderer` through an iframe and provide down the certificate for the display.
+- `Decentralised Renderer` or `Document renderer`: Website responsible of displaying one or more certificates. Usually they are not usable outside a iframe, but keep in mind it's a plain website.
+- `Layout`: A template used to display a specific certificate in a document renderer. A document renderer is able to rendere multiple kind of certificates. In other words, a document rednerer supports multiple `Layouts`
+- `View`: A `Layout` may be composed of multiples view. For instance in Opencerts, You may have a view to display the certificate, and a view to display the transcripts.
+- `Action`: A plain Javascript object used for the communication between the document viewer and the document renderer
+
```

Comment:
ah yes, these definitions will definitely be helpful.
minor spelling mistake at line 22. 
`In other words, a document rednerer supports multiple `Layouts``

Reply:
done